docs/master/api-reference/directives.md

```diff
@@ -2867,57 +2864,79 @@ Install the dependency [mll-lab/graphql-php-scalars](https://github.com/mll-lab/
 
     composer require mll-lab/graphql-php-scalars
 
-It contains the scalar type `Mixed`, which enables the dynamic query capabilities.
+### Usage
 
 ```graphql
-scalar Mixed @scalar(class: "MLL\\GraphQLScalars\\Mixed")
+type Person {
+    id: ID!
+    age: Int!
+    height: Int!
+    type: String!
+    hair_colour: String!
+}
+
+type Query {
+    people(
+        where: WhereConstraints @whereConstraints(columns: ["age", "type", "haircolour", "height"])
+    ): [Person!]! @all
+}
 ```
 
-Add an enum type `Operator` to your schema. Depending on your
-database, you may want to allow different internal values. This default
-should work for most databases:
+Lighthouse generates definitions for an `Enum` type and an `Input` type
+that are restricted to the defined columns.
 
 ```graphql
-enum Operator {
-    EQ @enum(value: "=")
-    NEQ @enum(value: "!=")
-    GT @enum(value: ">")
-    GTE @enum(value: ">=")
-    LT @enum(value: "<")
-    LTE @enum(value: "<=")
-    LIKE @enum(value: "LIKE")
-    NOT_LIKE @enum(value: "NOT_LIKE")
+"Dynamic WHERE constraints for the `where` argument on the query `people`.
+input PeopleWhereWhereConstraints {
+    column: PeopleWhereColumn
+    operator: SQLOperator = EQ
+    value: Mixed
+    AND: [PeopleWhereWhereConstraints!]
+    OR: [PeopleWhereWhereConstraints!]
+    NOT: [PeopleWhereWhereConstraints!]
```

Comment:
I assumed the duplicated word `Where` was a typo in #1119 and changed it to `PeopleWhereConstraints`. Is that correct?

Reply:
The `WhereConstraints` part at the end is static, `People` is based on the Type, and `Where` is based on the field name. So I assume it's not a typo, it's just that the field is named `where`.

Reply:
We might simplify this by moving `@whereConstraints` to the field level and only ever having a single generated argument, see https://github.com/nuwave/lighthouse/issues/1121